Common/Gfx/Canvas.cpp

```diff
@@ -563,6 +570,29 @@ void Canvas::DrawMaskedBitmap(Gdiplus::Bitmap* bitmap, Gdiplus::Bitmap* maskBitm
 	bitmapLock->Release();
 }
 
+void Canvas::DrawGeometry(const Shape & shape, D2D1_MATRIX_3X2_F & transform)
+{
+	if (!BeginTargetDraw()) return;
+
+	m_Target->SetAntialiasMode(D2D1_ANTIALIAS_MODE_ALIASED);
+	D2D1_MATRIX_3X2_F worldTransform;
+	m_Target->GetTransform(&worldTransform);
+	m_Target->SetTransform(transform * worldTransform);
+	Microsoft::WRL::ComPtr<ID2D1SolidColorBrush> solidBrush;
+	HRESULT hr = m_Target->CreateSolidColorBrush(shape.m_FillColor, solidBrush.GetAddressOf());
+	if (SUCCEEDED(hr)) {
+		if (shape.m_FillColor.a > 0)
+			m_Target->FillGeometry(shape.m_Shape.Get(), solidBrush.Get());
+
+		solidBrush->SetColor(shape.m_OutlineColor);
+		if (shape.m_OutlineColor.a > 0)
+			m_Target->DrawGeometry(shape.m_Shape.Get(), solidBrush.Get(), shape.m_OutlineWidth);
+	}
+	solidBrush.Reset();
```

Comment:
You should use `ReleaseAndGetAddressOf()` instead of `GetAddressOf()` in that case.